src/vsr/replica.zig

```diff
@@ -7918,36 +7911,29 @@ pub fn ReplicaType(
             assert(self.grid_repair_tables.executing() == 0);
             assert(self.grid_repair_writes.executing() == 0);
             assert(self.grid.blocks_missing.faulty_blocks.count() == 0);
+
             if (self.status == .normal) assert(!self.primary());
 
-            const stage: *const SyncStage.RequestingTrailers = &self.syncing.requesting_trailers;
-            assert(stage.done());
+            const stage: *const SyncStage.RequestingCheckpoint =
+                &self.syncing.requesting_checkpoint;
+            assert(stage.target.checkpoint_id == vsr.checksum(std.mem.asBytes(checkpoint_state)));
 
-            log.debug("{[replica]}: sync_requesting_trailers_callback: " ++
+            log.debug("{[replica]}: sync_requesting_checkpoint_callback: " ++
                 "checkpoint_op={[checkpoint_op]} checkpoint_id={[checkpoint_id]x:0>32} " ++
                 "manifest_oldest_checksum={[manifest_oldest_checksum]} " ++
                 "manifest_oldest_address={[manifest_oldest_address]} " ++
                 "manifest_newest_checksum={[manifest_newest_checksum]} " ++
-                "manifest_newest_address={[manifest_newest_address]} " ++
-                "client_sessions_checksum={[client_sessions_checksum]x:0>32}", .{
+                "manifest_newest_address={[manifest_newest_address]} ", .{
                 .replica = self.replica,
                 .checkpoint_op = stage.target.checkpoint_op,
                 .checkpoint_id = stage.target.checkpoint_id,
                 .manifest_oldest_checksum = checkpoint_state.manifest_oldest_checksum,
                 .manifest_oldest_address = checkpoint_state.manifest_oldest_address,
                 .manifest_newest_checksum = checkpoint_state.manifest_newest_checksum,
                 .manifest_newest_address = checkpoint_state.manifest_newest_address,
-                .client_sessions_checksum = stage.trailers.get(.client_sessions).final.?.checksum,
             });
 
-            inline for (comptime std.enums.values(vsr.SuperBlockTrailer)) |trailer| {
-                const trailer_size = stage.trailers.get(trailer).final.?.size;
-                @field(self.superblock, @tagName(trailer)).reset();
-                @field(self.superblock, @tagName(trailer)).decode(
-                    self.superblock.trailer_buffer(trailer)[0..trailer_size],
-                );
-            }
-            // Faulty bits will be set in state_machine_open_callback().
+            // Faulty bits will be set in sync_content().
```

Comment:
This comment was out of date. I think `state_machine_open_callback` did originally handle this logic, but it was moved to `sync_content` as soon as client-replies + table sync was added.